templates/layout-vertical.html.twig

```diff
@@ -165,3 +151,22 @@ Enjoy your theme!
 {% endblock %}
 </body>
 </html>
+
+{% macro navbar(navbar_start, navbar_notifications, navbar_user, navbar_end) %}
+    {% block navbar_start %}{{ navbar_start ?? '' | raw }}{% endblock %}
```

Comment:
Twig don't allow to create multiple blocks with the same name.
As we discuss already, `if/else` twig condition, are not ignored when rendering template.

So it ended as a `The block 'navbar_start' has already been defined line x`.

This is the only work around I've found..

Reply:
I have never seen blocks inside a macro, what is the difference when you remove them completely and only leave the content inside?

Reply:
> what is the difference when you remove them completely and only leave the content inside?

I have a slow brain during week-end, what do you mean by that, I don't understand.

Reply:
I mean: what happens when you have this code
```twig
{% macro navbar(navbar_start, navbar_notifications, navbar_user, navbar_end) %}
    {{ navbar_start ?? '' | raw }}
        {% if navbar_notifications is defined %}
            {{ navbar_notifications|raw }}
        {% else %}
            {% include '@Tabler/includes/navbar_notifications.html.twig' %}
        {% endif %}
        {% if navbar_user is defined %}
            {{ navbar_user|raw }}
        {% else %}
            {% include '@Tabler/includes/navbar_user.html.twig' %}
        {% endif %}
    {{ navbar_end ?? '' | raw }}
{% endmacro %}
```

Reply:
> I have never seen blocks inside a macro

ATE, it is not really a `block`, only a string.
So the variables given to the macro are strings.
This is why `xxx | raw` filter is used, to convert string into pure HTML.

Reply:
> what is the difference when you remove them completely and only leave the content inside?

~I have a slow brain during week-end, what do you mean by that, I don't understand.~

I get it now.

You mean if the macro would look like this.

```twig
{% macro navbar(navbar_start, navbar_notifications, navbar_user, navbar_end) %}
    {{ (navbar_start ?? '') | raw }}
        {% if navbar_notifications is defined %}
            {{ navbar_notifications|raw }}
        {% else %}
            {% include '@Tabler/includes/navbar_notifications.html.twig' %}
        {% endif %}
        {% if navbar_user is defined %}
            {{ navbar_user|raw }}
        {% else %}
            {% include '@Tabler/includes/navbar_user.html.twig' %}
        {% endif %}
    {{ (navbar_end ?? '') | raw }}
{% endmacro %}
```

It will be the same result.

Personally, I let block inside this kind of macro.
It allow me to have a friendly reminder that I can override blocks from extends.

We can remove it if you want.

Reply:
I also like having the reminder there.

The includes can be done as a function, which is the recommended way, e.g.

```
 {{ include '@Tabler/includes/navbar_user.html.twig' }}
```

Reply:
@tacman twig doesn't allow blocks override from extend, embedded from inside, include or use